### PR TITLE
Added functionality for AutocompleteInput to match any part of the input item

### LINKED ISSUE #11571: [FEATURE] Make AutocompleteInput more usable by providing an option to match any part of the input items
**Problem:**
AutocompleteInput only matches start of the items, this severely limits the usability in some use cases.
e.g. when list is  ['Biology', 'Chemistry', 'Physics']
Typing "mis" should match Chemistry
Now it only matches when typing "Che"

**Solution:**
Provide a new option to enable matching any part of the string not just the beginning. \
The Javascript changes are trivial (it now uses string function StartsWith)

### PR DESCRIPTION
**Pull request by jennazch**
Fixes #11571 

Typing any part of the query will autocomplete the word. For example, when the list is ["Chemistry"], typing in "mis" will also lead autocomplete to suggest the full word "Chemistry."

### CODE REVIEW

**Comment by bryevdv** [comment]
Hi @jennazch it looks like there is something weird going on with the commit history, and there are some unrelated changes being brought in. Here are some things you could try:

* Do an [interactive rebase](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history#interactive-rebase) on `branch-3.0` and remove the 8b4270d commit that seems to have the spurious changes
* Open a new PR branched of latest `branch-3.0` with just the relevant changes 

In any case you will want to make sure `branch-3.0` is locally update to date with the latest, first (i.e. do a `git pull` on `branch-3.0`)

Once that is sorted, just to give an overview, the remaining work for the PR to be complete will be:

* add a new property (Python and JS sides) to control what kind of matching should be done
* condition the matching codepaths on that property value (i.e. default to current "startswith" matching and only do "incudes" matching if the user asked for it)
* Add a test for this (we can help with that when things are further along)

File: bokehjs/src/lib/models/widgets/autocomplete_input.ts [hunk 1]
```diff
@@ -149,8 +149,10 @@ export class AutocompleteInputView extends TextInputView {
           if (acnorm(text).startsWith(acnorm(value))) {
             completions.push(text)
           }
+          if (acnorm(text).includes(acnorm(value))) {
+            completions.push(text)
+          }
```
Comment (mattpap) [inline bokehjs/src/lib/models/widgets/autocomplete_input.ts:154]: Just adding this makes the previous clause pointless. Either the previous one should be removed, or the behavior should be made configurable, by adding a new property to `AutocompleteInput` model. The later seems a better idea, as we may introduce other matching algorithms in future (e.g. some sort of fuzzy matching).

Comment (bryevdv) [reply bokehjs/src/lib/models/widgets/autocomplete_input.ts:154]: A configurable property is definitely needed, see https://github.com/bokeh/bokeh/pull/11889#issuecomment-992797211

**Comment by bryevdv** [comment]
@jennazch will you be able to continue on this Pull Request? If not, we can close it

**Comment by bryevdv** [comment]
Closing for lack of response/activity @jennazch if you are able to return to this in the future we are happy to look at a new PR

**Pull request closed without merging**